src/Lab/Experiments/ImGuiVulkan/ImGuiVulkanApplication.cs

```diff
@@ -320,12 +321,20 @@ private unsafe void CreateInstance()
 		{
 			_vk = Vk.GetApi();
 
+            var isMacOs = Environment.OSVersion.Platform is PlatformID.Unix or PlatformID.MacOSX; 
```

Comment:
This seems wrong.

```suggestion
            var isMacOs = Environment.OSVersion.Platform is PlatformID.MacOSX; 
```

surely?

Reply:
Indeed, the problem is that MacOSX is showing like Unix in my system, I don't know why, it would need some testing on Linux to see how it behaves or may be a bug in the runtime.

Reply:
Can we use OperatingSystem.IsMacOS() instead given you've retargeted to .NET 6? These are the latest APIs that should work.

Reply:
I would suggest solving this right now to remove the Unix part to avoid issues on the Linux side and place a comment over the assignment warning the developers of the issue.
I could follow up with the runtime team to find out what is going on.

Reply:
I think the old Environment/RuntimeEnvironment variables are deprecated (and not really guaranteed to work) anyway, so the runtime team would probably just point you at `OperatingSystem`.

Reply:
Fixed now with the solution proposed here:
https://github.com/dotnet/runtime/issues/21660#issuecomment-633731507